Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py

```diff
@@ -217,54 +434,93 @@ def fetch_indicators_command(client: Client) -> List[Dict[str, Any]]:  # pragma:
-            if len(indicators) < 15000 and next:
-                url = next.replace(demisto.getParam('tc_api_path'), '')
-            else:
-                break
+        response, _, next_link = client.make_request(Method.GET, url, get_next=True)
+        indicators.extend(response)
+        # Limit the number of results to not get an error from the API
+        if len(indicators) < 15000 and next_link:
+            url = next_link.replace(demisto.getParam('tc_api_path'), '')
+            url = url.replace('+', '%20')
+
+        else:
+            break
 
     return indicators
 
 
+def set_tql_query(from_date):
+    params = demisto.params()
+    owners = f'AND ({create_or_query("ownerName", params.get("owners"))}) '
+    tags = f'AND ({create_or_query("tags", params.get("tags"))}) '
+    status = f'AND ({create_or_query("status", params.get("status"))}) '
+    active_only = 'AND indicatorActive EQ True ' if argToBoolean(params.get("indicatorActive")) else ''
+    confidence = f'AND confidence GT {params.get("confidence")} ' if params.get("confidence") else ''
+    threat_score = f'AND threatAssessScore GT {params.get("threatAssessScore")} ' if params.get("threatAssessScore") else ''
+
+    type_name_query = create_types_query()
+    type_names = f'AND {type_name_query}' if type_name_query else ''
+
+    tql = f'{owners if owners != "AND () " else ""}' \
+          f'{tags if tags != "AND () " else ""}' \
+          f'{from_date if from_date != "AND () " else ""}' \
+          f'{status if status != "AND () " else ""}' + active_only + confidence + threat_score + type_names
+
+    tql = tql.replace('AND ', '', 1)
+    return tql
+
+
 def get_indicators_command(client: Client):  # pragma: no cover
     """ Get indicator from ThreatConnect, Able to change limit and offset by command arguments.
-
     Args:
         client: ThreatConnect client.
-
     Returns:
         str: Human readable.
         dict: Operation entry context.
         dict: Operation raw response.
     """
-    limit = demisto.args().get('limit', '50')
-    offset = demisto.args().get('offset', '0')
-    owners = demisto.getArg('owners') or demisto.getParam('owners')
-    owners = create_or_query(owners, "ownerName")
-    owners = urllib.parse.quote(owners.encode('utf8'))  # type: ignore
-    url = f'/api/v3/indicators?tql={owners}&resultStart={offset}&resultLimit={limit}'
+    args = demisto.args()
+    limit = args.get('limit', '50')
+    offset = args.get('offset', '0')
+
+    owners = f'AND ({create_or_query("ownerName", args.get("owners"))}) ' if args.get("owners") else ''
+    active_only = f'AND indicatorActive EQ {args.get("indicatorActive")} ' if argToBoolean(args.get("indicatorActive")) else ''
+    confidence = f'AND confidence GT {args.get("confidence")} ' if args.get("confidence") else ''
+    threat_score = f'AND threatAssessScore GT {args.get("threatAssessScore")} ' if args.get("threatAssessScore") else ''
```

Comment:
same here

Reply:
Checked with Danil, there is no need to use the other parameter when the arguments are absent